### PR TITLE
feat: [AXIMST-578] display outline sidebar depends on feature flag

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** display outline sidebar depends on feature flag

**Youtrack:** https://youtrack.raccoongang.com/issue/AXIMST-578

### CODE REVIEW

File: src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      Icon = isCompleted ? BookCompleteIcon : BookIcon;
+  }
+
+  return (
+    <Icon {...props} className={classNames({ 'text-success': isCompleted, 'text-gray-700': !isCompleted })} />
+  );
+};
+
+UnitIcon.propTypes = {
+  type: PropTypes.oneOf(['video', 'other', 'vertical', 'problem', 'lock']).isRequired,
```
Comment (PKulkoRaccoonGang) [inline src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx:42]: [question]: Can we refactor this file and create an additional constant object? For example, we have a similar use of icon types in [another file](https://github.com/raccoongang/frontend-app-learning/blob/4797425a215ce7b81b539489d048f809b03639da/src/courseware/course/sequence/sequence-navigation/UnitIcon.jsx#L11) of this repository ['video', 'other', 'vertical', 'problem', 'lock'].

For example:
`const UNIT_ICON_TYPES = { video: 'video', other: 'other' ... }`

and then:
`case UNIT_ICON_TYPES.video:`

...

`type: PropTypes.oneOf(Object.keys(UNIT_ICON_TYPES)).isRequired,`

**Review by PKulkoRaccoonGang — CHANGES_REQUESTED** [review]

File: src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx (hunk 1 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx:42]: Changed in https://github.com/raccoongang/frontend-app-learning/pull/18

File: src/courseware/data/thunks.js [hunk 2]
```diff
@@ -6,8 +6,9 @@ import {
 import {
   getBlockCompletion,
   getCourseDiscussionConfig,
-  getCourseMetadata,
+  getCourseMetadata, getCourseOutline,
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/data/thunks.js:9]: Fixed

File: src/courseware/data/thunks.js [hunk 3]
```diff
@@ -30,18 +35,25 @@ export function fetchCourse(courseId) {
...
       getCourseHomeCourseMetadata(courseId, 'courseware'),
+      getCoursewareOutlineSidebarEnabledFlag(courseId),
     ]).then(([
       courseMetadataResult,
       learningSequencesOutlineResult,
-      courseHomeMetadataResult]) => {
-      if (courseMetadataResult.status === 'fulfilled') {
+      courseHomeMetadataResult,
+      courseOutlineSidebarDisableFlagResult]) => {
+      const fetchedMetadata = courseMetadataResult.status === 'fulfilled';
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/courseware/data/thunks.js:44]: It's a correct code formatting

**Review by monteri — APPROVED** [review]

**Pull request merged**